File: CS50/pset6/caesar/caesar.py
import sys
from cs50 import get_string


def main():
    if len(sys.argv) == 2:
        n = int(sys.argv[1])
        s = get_string("plaintext: ")
        caesar = []
        # Encrypting
        for k in s:
            # Making sure caption is ok
            if k.isupper():
                c = ((ord(k) - ord("A") + n) % 26) + ord("A")
                caesar.append(chr(c))
            elif k.islower():
                c = ((ord(k) - ord("a") + n) % 26) + ord("a")
                caesar.append(chr(c))
            else:
                caesar.append(k)
        print("ciphertext: ", end='')
        print("".join(caesar))
        exit(0)
    else:
        print("Usage: ./caesar k")
        exit(1)


if __name__ == "__main__":
    main()

File: CS50/pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])

{
    if (argc == 2)
    {
        string str = argv[1];
        int n = atoi(str);
        string s = get_string("plaintext: ");
        //Encrypting
        int i = strlen(s);
        int k;
        for (k = 0; k < i; k++)
        {
            //Making sure caption is ok
            if (s[k] >= 'A' && s[k] <= 'Z')
            {
                s[k] = ((s[k] - 'A' + n) % 26) + 'A';
            }
            else if (s[k] >= 'a' && s[k] <= 'z')
            {
                s[k] = ((s[k] - 'a' + n) % 26) + 'a';
            }
            else
            {
                s[k] = s[k];
            }
        }

        printf("ciphertext: %s\n", s);
    }
    else
    {
        printf("Usage: ./caesar k");
        return 1;
    }
}